gnosis/safe/safe.py

```diff
@@ -519,16 +522,28 @@ def contract(self) -> Contract:
 
     @cached_property
     def domain_separator(self):
-        return fast_keccak(
-            encode_abi(
-                ["bytes32", "uint256", "address"],
-                [
-                    self.DOMAIN_TYPEHASH,
-                    self.ethereum_client.get_chain_id(),
-                    self.address,
-                ],
+        version = self.contract.functions.VERSION().call()
```

Comment:
I am following the implementation of `domain_separator` which is computed instead of calling the contract.
I'm not sure why, but maybe this was implemented at this way to avoid asking the blockchain one more time.
If this were the reason, then this implementation should be changed to create a `@cached_property` of the contract version or perhaps call `domain_separator` directly to the blockchain instead of requesting the version once more.

Reply:
I'd say that the library was just assuming 1.3.0 for the domain separator. Therefore, it wouldn't perform any RPC calls given that assumption.